tests/testTriggers.go

```diff
@@ -3735,6 +3763,178 @@ func TriggerPoolAddDiskAndReboot(contexts *[]*scheduler.Context, recordChan *cha
+				UpdateOutcome(event, err)
+				//Removing the label if autopilot rule creation is failed
+				if err != nil {
+					for k := range poolLabel {
+						Inst().S.RemoveLabelOnNode(autLabelNode, k)
+						autLabelNode = node.Node{}
+					}
+				}
+			} else {
+				logrus.Warn("Skipping autopilot rule creation as node is not labelled")
+			}
+
+		})
+	} else {
+
+		Step("validate the  autopilot events", func() {
+			logrus.Infof("validate the  autopilot events for %s", apRule.Name)
+			ruleEvents, err := core.Instance().ListEvents("", meta_v1.ListOptions{
+				FieldSelector: fmt.Sprintf("involvedObject.kind=AutopilotRule,involvedObject.name=%s", apRule.Name),
+			})
+			UpdateOutcome(event, err)
+
+			if err == nil {
+				for _, ruleEvent := range ruleEvents.Items {
+					//checking the events triggered in last 60 mins
+					if ruleEvent.LastTimestamp.Unix() < meta_v1.Now().Unix()-3600 {
+						continue
+					}
+					logrus.Infof("autopilot rule event reason : %s and message: %s ", ruleEvent.Reason, ruleEvent.Message)
+					if strings.Contains(ruleEvent.Reason, "FailedAction") {
+						UpdateOutcome(event, fmt.Errorf("autopilot rule %s failed, Reason: %s, Message; %s", apRule.Name, ruleEvent.Reason, ruleEvent.Message))
+					}
+
+				}
+
+			}
+
+		})
+
+		Step("validate Px on the rebalanced node", func() {
+			logrus.Infof("Validating PX on node : %s", autLabelNode.Name)
+			err := Inst().V.WaitDriverUpOnNode(autLabelNode, 1*time.Minute)
+			UpdateOutcome(event, err)
+
+			rebalanceJobs, err := Inst().V.GetRebalanceJobs()
+			UpdateOutcome(event, err)
+			if err == nil {
+
+				for _, job := range rebalanceJobs {
+					jobResponse, err := Inst().V.GetRebalanceJobStatus(job.GetId())
+					UpdateOutcome(event, err)
+					if err == nil {
+
+						previousDone := uint64(0)
+						jobState := jobResponse.GetJob().GetState()
+						if jobState == opsapi.StorageRebalanceJobState_CANCELLED {
+							UpdateOutcome(event, fmt.Errorf("job %v has cancelled, Summary: %+v", job.GetId(), jobResponse.GetSummary().GetWorkSummary()))
+
+						}
+
+						if jobState == opsapi.StorageRebalanceJobState_PAUSED || jobState == opsapi.StorageRebalanceJobState_PENDING {
+							logrus.Infof("Job %v is in paused/pending state", job.GetId())
+						}
+
+						if jobState == opsapi.StorageRebalanceJobState_DONE {
+							logrus.Infof("Job %v is in DONE state", job.GetId())
+						}
+
+						if jobState == opsapi.StorageRebalanceJobState_RUNNING {
+							logrus.Infof("Job %v is in Running state", job.GetId())
+
+							currentDone, total := getReblanceWorkSummary(jobResponse)
+							//checking for rebalance progress
+							for currentDone < total && previousDone < currentDone {
+								time.Sleep(2 * time.Minute)
+								logrus.Infof("Waiting for job %v to complete current state: %v, checking again in 2 minutes", job.GetId(), jobState)
+								jobResponse, err = Inst().V.GetRebalanceJobStatus(job.GetId())
+								UpdateOutcome(event, err)
+								if err == nil {
```

Comment:
we can use `if err != nil then break` and remove the else part

Reply:
fixed